virttest/test_setup.py

```diff
@@ -1470,53 +1470,46 @@ def sr_iov_setup(self):
         if process.system("lsmod | grep -w %s" % self.driver, ignore_status=True,
                           shell=True):
             re_probe = True
-        # If driver is available then set VFs for ppc64le
-        else:
-            if ARCH == 'ppc64le' and self.driver == 'mlx5_core':
-                pf_devices = self.get_pf_mlx()
-            elif ARCH == 'x86_64':
-                pf_devices = self.get_pf_ids()
+        # If driver is available and pa_type is vf then set VFs
+        elif(self.pa_type == 'vf'):
+            pf_devices = self.get_pf_ids()
             for PF in pf_devices:
                 if not self.set_vf(PF, self.driver_option):
                     re_probe = True
             if not self.check_vfs_count():
                 re_probe = True
-            if not re_probe:
-                self.setup = None
-                return True
+        if not re_probe:
+            self.setup = None
+            return True
 
         # Re-probe driver with proper number of VFs once more and raise
         # exception
         if re_probe:
             if not self.remove_driver() or not self.modprobe_driver():
                 return False
-            if ARCH == 'ppc64le' and self.driver == 'mlx5_core':
-                set_ip = 0
-                if (self.static_ip):
-                    if (not self.start_addr_PF) or (not self.net_mask):
-                        raise exceptions.TestSetupFail(
-                            "No IP / netmask found, please populate starting IP address for PF devices in configuration file")
-                    ip_addr = utils_net.IPAddress(self.start_addr_PF)
-                    set_ip = 1
-                for PF in pf_devices:
-                    if set_ip:
-                        ip_assign = "ifconfig %s %s netmask %s up" % (
-                            PF, ip_addr, self.net_mask)
-                        logging.info("assign IP to PF device %s : %s", PF,
-                                     ip_assign)
-                        if process.system(ip_assign, shell=True,
-                                          ignore_status=True):
-                            return False
-                        ip_addr = ip_addr + 1
-                    logging.info("PF device '%s'.", PF)
-                    if not self.set_vf(PF, self.driver_option):
-                        return False
-            elif ARCH == 'x86_64':
-                pf_devices = self.get_pf_ids()
-                for PF in pf_devices:
-                    if not self.set_vf(PF, self.driver_option):
+            set_ip = 0
+            if (self.static_ip):
+                if (not self.start_addr_PF) or (not self.net_mask):
+                    raise exceptions.TestSetupFail(
+                        "No IP / netmask found, please populate starting IP address for PF devices in configuration file")
+                ip_addr = utils_net.IPAddress(self.start_addr_PF)
```

Comment:
@huangyum  could you explain this line ?

Reply:
I just kept it as it was since I saw other pr (https://github.com/avocado-framework/avocado-vt/pull/1139) handling this part. Thanks!

Reply:
@huangyum  i can not understand it well, evenif you follow the design from other PR, i think it is needed to make it clearly.  thx!

Reply:
OK. I keep it the original way for now. Please help review again, thanks!